src/hooks/useAccountId.ts

```diff
@@ -0,0 +1,52 @@
+import { useEffect, useState, useCallback, SetStateAction } from "react";
+import { AccountId } from "@hashgraph/sdk";
+import { hashConnectWallet } from "@/services/wallets/hashconnect/hashconnectClient";
+
+type FormValues = { [key: string]: string };
+
+export function useAccountId<T>(
+  setValues: SetStateAction<any>,
+  values: T,
+  columnName: string,
+) {
+  const [hederaAccountIdError, setHederaAccountIdError] = useState(false);
+
+  const handleAddressValueChange = useCallback(async () => {
+    const address = (values as FormValues)[columnName];
```

Comment:
I have couple of concerns with current setup.

- the hook feels a bit tied up to the form itself. can we make it a bit more pure, with just "passing address in, getting address out (or error)" and deal with updating the form itself separately, in the component itself?

- I feel it might be not optimal UI for the user when he pastes something in the input and it gets rewritten right away. I'd suggest keeping the conversion behind the scenes, or maybe just show the converted value below the input for information purpose, and use it when submitting the form, wdyt?

Reply:
addition to the point N1, hook might not update the form at all, but just store and return the converted value, and we grab the value when submitting the form

Reply:
> addition to the point N1, hook might not update the form at all, but just store and return the converted value, and we grab the value when submitting the form

Makes sense for me, will do that.